### ISSUE TITLE
Issue with authenticating between Canva and MIRO

### ISSUE THREAD

**Issue by brooksr4** [blueorbitz/canva-for-miro-issues#1]
I was trying to set up Canva for MIRO and when I went to set up the authentication on the Canva side I got the following error message:
Application error: a client-side exception has occurred (see the browser console for more information).

Here is a screenshot of the error message and the Console Log:
<img width="1678" alt="Screenshot 2024-10-09 at 11 22 50 PM" src="https://github.com/user-attachments/assets/35d86b61-1af6-4071-9f59-30ff2b1b3aef">

**Comment by brooksr4** [comment]
Actually I fixed the error by doing it in the same browser. I typically use the MIRO Desktop version. That's where the issue lies.

**Comment by blueorbitz** [comment]
Hi @brooksr4, thanks for trying out the app. I appreciate the feedback.  Indeed, the desktop version was not in consideration.
I have downloaded the desktop version too, and I am able to replicate the issue.

What actually happened is when you try to authenticate the app, it redirects to a new window to sign in to the app. However, the Miro's desktop app isn't able to pick the event sent from the browser. Therefore, it stuck at the loading screen.

I have not researched much on how to fix this. Might need a bit of time to tackle such an issue. In the meantime, appreciate if you could continue to use the app from the browser at the moment.

**Comment by blueorbitz** [comment]
Hi @brooksr4, it seems that the browser postMessage isn't able to communicate the sign-in code with the Desktop app.

Therefore, as an alternative solution. I decided to make it easy for you to copy the required string to complete the login process.

Try clicking on the "Manually paste code instead" to switch to the form view. I believe the rest should be straight-forward following the form hints.

Here are some screenshots of what I have changed.

![image](https://github.com/user-attachments/assets/cd05bead-e6de-4b2e-8cd1-67514b012b85)

![image](https://github.com/user-attachments/assets/cb9ac6fe-7309-4f59-b933-4afabfdcbe13)
